mapie/classification.py

```diff
@@ -830,190 +835,192 @@ def predict(
 
         Returns
         -------
-        Union[ArrayLike, Tuple[ArrayLike, ArrayLike]]
+        Union[NDArray, Tuple[NDArray, NDArray]]
 
-        - ArrayLike of shape (n_samples,) if alpha is None.
+        - NDArray of shape (n_samples,) if alpha is None.
 
-        - Tuple[ArrayLike, ArrayLike] of shapes
+        - Tuple[NDArray, NDArray] of shapes
         (n_samples,) and (n_samples, n_classes, n_alpha) if alpha is not None.
         """
         if self.method == "top_k":
             agg_scores = "mean"
         # Checks
         cv = check_cv(self.cv)
         include_last_label = self._check_include_last_label(include_last_label)
-        alpha_ = check_alpha(alpha)
+        alpha = cast(Optional[NDArray], check_alpha(alpha))
         check_is_fitted(self, self.fit_attributes)
         if self.image_input:
             check_input_is_image(X)
 
         # Estimate prediction sets
         y_pred = self.single_estimator_.predict(X)
-        n = self.n_samples_
-        if alpha_ is None:
+        n = len(self.conformity_scores_)
+
+        if alpha is None:
             return np.array(y_pred)
 
+        # Estimate of probabilities from estimator(s)
+        # In all cases : len(y_pred_proba.shape) == 3
+        # with  (n_test, n_classes, n_alpha or n_train_samples)
+        alpha_np = cast(NDArray, alpha)
```

Comment:
Wasn't `alpha` converted to a `NDArray` line 850 ?

Reply:
No, `alpha` was cast to `Optional[NDArray]`, so that checking against `None` does satisfy mypy

Reply:
`mypy` is really picky !